--- iptraf-ng-1.1.3.1/src/ipfilter.c.formatsec	2013-02-05 13:57:52.461008094 +0100
+++ iptraf-ng-1.1.3.1/src/ipfilter.c	2013-02-05 13:58:17.100874609 +0100
@@ -136,7 +136,7 @@ void gethostparams(struct hostparams *da
 				snprintf(msgstr, 60,
 					 "Invalid protocol input at or near token \"%s\"",
 					 bptr);
-				tui_error(ANYKEY_MSG, msgstr);
+				tui_error(ANYKEY_MSG, "%s", msgstr);
 				doagain = 1;
 			} else
 				doagain = 0;
--- iptraf-ng-1.1.3.1/src/othptab.c.formatsec	2013-02-05 14:02:18.220828537 +0100
+++ iptraf-ng-1.1.3.1/src/othptab.c	2013-02-05 14:09:36.732870661 +0100
@@ -430,7 +430,7 @@ void printothpentry(struct othptable *ta
 				break;
 			}
 
-			sprintf(scratchpad, rarp_mac_addr);
+			sprintf(scratchpad, "%s", rarp_mac_addr);
 			strcat(msgstring, scratchpad);
 			wattrset(table->othpwin, ARPATTR);
 			break;
@@ -505,7 +505,7 @@ void printothpentry(struct othptable *ta
 		wattrset(table->othpwin, UNKNIPATTR);
 		protptr = getprotobynumber(entry->protocol);
 		if (protptr != NULL) {
-			sprintf(protname, protptr->p_aliases[0]);
+			sprintf(protname, "%s", protptr->p_aliases[0]);
 		} else {
 			sprintf(protname, "IP protocol");
 			unknown = 1;
